python/src/scipp/plotting/model.py

```diff
@@ -198,6 +202,57 @@ def _string_tick_formatter(self, array_values, size):
         return lambda val, pos: array_values[int(val)] if (int(
             val) >= 0 and int(val) < size) else ""
 
+    def _date_tick_formatter(self, offset, dim):
+        """
+        Format datetime ticks: adjust the time precision and update
+        offset according to the currently displayed range.
+        """
+        def formatter(val, pos):
+            dt = str(np.datetime64(int(val) + int(offset), 'ns'))
```

Comment:
Above you said you need an (interger?) offset, but wouldn't this be simpler here if `offset = sc.min(coord)`? Then here it might be something like
```python
dt = str(offset + val)
```

Reply:
i need the `np.datetime64(..., 'ns')` to convert to a human readable string.
and the explicit `int()` conversions in here are apparently needed even if offset (i thought) was an `int`, but i think it was a numpy int, and this fails to convert to datetime (this was what took me ages to figure out, and I should indeed make a note of this in a comment).

try
```Py
a = np.arange(10)
np.datetime64(a[5], 'ns')
```
and you'll get `ValueError: Could not convert object to NumPy datetime`, while
```Py
np.datetime64(5, 'ns')
```
works

Reply:
`offset` should be `sc.dtype.datetime64`, then `str` should work. Why do you need the numpy datetime formatting rather than scipp's?

Reply:
```python
# above
offset = sc.min(coord)
coord = coord - offset # int
# here
dt = str(offset + val) # offset + integer val = datetime
```

Reply:
This is a bit strange. An offset should be a difference type, i.e. int here, coord should be a datetime and so should val (after conversion from matplotlib anyway).
There is no `sc.dtype.datetime64` object in Python, all datetimes get converted to `np.datetime64` unless you want to use a `sc.Variable` here. So unless you also want to use `np.timedelta64` for arithmetic, the easiest solution is to just stick with ints for intermediate steps.

Reply:
I think `offset` is just a misnomer. It is actually the lowest data in the range.

With my code `offset = sc.min(coord)` it is just a scalar variable. Actually the last line should be `dt = str((offset + val).value)`.

Reply:
So with mpl, `val` is a always float. It's the value of x, where the tick mark is being drawn.
mpl tries to keep the number of ticks it generates more or less constant, and so even if your coordinate is integers, `val` can be floats that are not rounded integers, when you zoom in enough.

Doing
```Py
time = sc.array(dims=['time'], values=np.arange(np.datetime64('2017-01-01T12:00:00'),
                                                np.datetime64('2017-01-01T13:00:00')))
```
will give you an array of 3600 datetimes separated by 1s.
Doing `dt = time - sc.min(time)` gives you integers, from 0 to 3599.
So if you are converting `val` to an `int` inside the formatter, if you zoom in far enough (so that you only see one data point for example), several of the values you see on the ticks will be the same, because they round up to the same integer.

One could maybe say that it ok, because you shouldn't probably be zooming that far in anyway, since you won't be displaying a meaningful view anyway (with only one or no data points in the field of view).
However, the formatter is actually also used to display the position of the mouse cursor at the bottom of the plot. And in principle, it might be useful to print in-between when hovering the cursor between points? I'm not sure.

So I converted everything to nanoseconds, because i considered that this would be enough time resolution for anything we try to display, even if we are still rounding to integer nanoseconds.

Maybe I should just try out your suggestion and see what it looks like.

Reply:
I guess a follow up question would be once I get my viewport bound as integers using scipp datetime, then I can know what unit they are but I then need to perform conversions when comparing to ns or us or day or year to decide what string should be on the tick and what should be in the offset string.

I would need to implement `to_unit ` for integers? Maybe that's ok?

Reply:
I see you made changes along that line, comparing `diff` to variables with a variety of time units? How does that work? I don't think we can or should support comparison operators with different units?

Reply:
Working on `to_unit` now, will open a PR later.

Reply:
Yes I started to make changes and then posted my question because it didn't work